builders/tslice.py

```diff
@@ -138,24 +135,40 @@ def build_df(
     if verify(input_yaml):
         # loop over datasets
         df_parts = []
+        entries = {}
```

Comment:
You might want to (at the top) import `from collections import defaultdict` and change this from a standard `dict` to a `defaultdict`:

```suggestion
        entries = defaultdict(dict)
```

The patterns you use below are simplified with a `defaultdict`.